manifests/proxy.pp

```diff
@@ -20,9 +20,14 @@
       mode    => '0644',
     }
 
-    Package['jenkins']
-    -> File[$proxy_xml]
-    ~> Class['jenkins::service']
+    if $jenkins::manage_service {
+      Package['jenkins']
+      -> File[$proxy_xml]
+      ~> Class['jenkins::service']
+    } else {
+      Package['jenkins']
+      -> File[$proxy_xml]
+    }
```

Comment:
What about splitting?

```suggestion
    Package['jenkins'] -> File[$proxy_xml]

    if $jenkins::manage_service {
      File[$proxy_xml] ~> Class['jenkins::service']
    }
```

Or maybe moving the `if $jenkins::manage_service` into the `jenkins::service` class and always notifying the class that may contain 0 resources? :shrug: